lib/manageiq/environment.rb

```diff
@@ -121,6 +121,46 @@ def self.bundler_version
       File.read(gemfile).match(/gem\s+['"]bundler['"],\s+['"](.+?)['"]/)[1]
     end
 
+    def self.warn_if_branch_out_of_date
+      return if %w(production test).include?(ENV['RAILS_ENV'].to_s)
+
+      Dir.chdir(APP_ROOT) do
+        system!("git fetch")
```

Comment:
Without any other options this will fetch from `origin` if there is one or the remote for the current branch which is probably the dev's fork.

If we're assuming `upstream` down [here](https://github.com/ManageIQ/manageiq/pull/18509/files#diff-7dbc55ccfbebe14a0aa870a17989c558R160) maybe do the same here?

Reply:
Yeah, I was trying to not assume upstream for the case where you have an upstream tracking branch.

I guess I can move this in the `upstream_tracking_branch.empty?` conditional below, like this:

```ruby
        commits =
          if upstream_tracking_branch.empty?
            system!("git fetch upstream")
            commits_behind_upstream
          else
            system!("git fetch")
            commits_behind_tracked_branch
          end
```